hail/src/main/scala/is/hail/expr/types/physical/PContainer.scala

```diff
@@ -318,15 +318,31 @@ abstract class PContainer extends PIterable {
     }
 
     val i = mb.newLocal[Long]
+    val mod = mb.newLocal[Int]
     Code(
-      i := PContainer.nMissingBytes(sourceType.loadLength(sourceOffset)).toL,
+      mod := (sourceType.loadLength(sourceOffset) % 8) - 1,
+      (mod >= 0).mux(
+        Code(
+          i := PContainer.nMissingBytes(sourceType.loadLength(sourceOffset)) - 1L,
+          Code.whileLoop(mod >= 0,
+            Code(
+              sourceType
+                .isElementMissing(sourceOffset, i.toI * 8 + mod)
```

Comment:
wasted multiplication, but runs a maximum of 7 times.

Reply:
the JIT will turn a `x * 8` into a `x << 3` on sight.